File: Práctica Grupal 2/practica2-grupo1-Equipo2/src/gestorAplicacion/clasesDeBiblioteca/Autor.py
class Autor:
    def __init__(self, nombre, nacionalidad, corriente):
        self.nombre = nombre
        self.nacionalidad = nacionalidad
        self.corriente = corriente
        self.obras = []

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_nacionalidad(self):
        return self.nacionalidad

    def set_nacionalidad(self, nacionalidad):
        self.nacionalidad = nacionalidad

    def get_corriente(self):
        return self.corriente

    def set_corriente(self, corriente):
        self.corriente = corriente

    def get_obras(self):
        return self.obras

    def set_obras(self, obras):
        self.obras = obras

    def __str__(self):
        return self.nombre

File: Práctica Grupal 1/src/gestorAplicacion/paquete2/Autor.java
package gestorAplicacion.paquete2;

import java.util.ArrayList;
import gestorAplicacion.paquete1.Libro;
import java.io.Serializable;

public class Autor implements Serializable{
	private static final long serialVersionUID = 1L;
    private String nombre; // Nombre del autor
    private String nacionalidad; // Nacionalidad del autor
    private String corriente; // Corriente literaria del autor
    private ArrayList<Libro> obras; // Array de obras escritas por el autor

    // Constructor de la clase Autor
    public Autor(String nombre, String nacionalidad, String corriente) {
        this.nombre = nombre;
        this.nacionalidad = nacionalidad;
        this.corriente = corriente;
        this.obras = new ArrayList<>();
    }
    
    public Autor() {
    	this("Autor anonimo", "Desconocido", "Desconocido");
    }

    // Método para obtener el nombre del autor
    public String getNombre() {
        return nombre;
    }

    // Método para establecer el nombre del autor
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    // Método para obtener la nacionalidad del autor
    public String getNacionalidad() {
        return nacionalidad;
    }

    // Método para establecer la nacionalidad del autor
    public void setNacionalidad(String nacionalidad) {
        this.nacionalidad = nacionalidad;
    }

    // Método para obtener la corriente literaria del autor
    public String getCorriente() {
        return corriente;
    }

    // Método para establecer la corriente literaria del autor
    public void setCorriente(String corriente) {
        this.corriente = corriente;
    }

    // Método para obtener la lista de obras escritas por el autor
    public ArrayList<Libro> getObras() {
        return obras;
    }

    // Método para establecer la lista de obras escritas por el autor
    public void setObras(ArrayList<Libro> obras) {
        this.obras = obras;
    }
    public String toString() {
    	return nombre;
    }
}
